perma_web/perma/utils.py

```diff
@@ -608,6 +612,21 @@ def clear_wr_session(request):
         logger.exception('Unexpected response from DELETE /user/{user}'.format(user=wr_username))
 
 
+def set_wr_uploaded(request, link):
+    """
+    Mark a link id/collection slug as having been uploaded
+    """
+    request.session['wr_uploaded:' + link] = True
+
+
+def get_wr_uploaded(request, link):
+    """
+    Mark a link id/collection slug as having been uploaded
+    """
+    return request.session.get('wr_uploaded:' + link)
+
+
+
```

Comment:
Great idea to store this info in the user's session!

I think the logic of these functions would fit quite naturally as methods on the `Link` model, rather than independent utility functions. E.g.
```
# models.py

class Link(DeletableModel):
    ...
    
    def set_uploaded_during_wr_session(self, request):
        """
        Record in user's session that this link has been uploaded to WR for playback
        """
        request.session['wr_uploaded:' + self.wr_collection_slug] = True

    def get_uploaded_during_wr_session(self, request):
        """
        Retrieve from user's session whether this link has been uploaded to WR for playback
        """
        return request.session.get('wr_uploaded:' + self.wr_collection_slug)
```
No strong opinions on method names 😉

Reply:
Sure, could move it there, but note that currently all WR related functions are in one place in utils, and also the `request.session` manipulation happens there. The cleanup `wr_uploaded` keys for example is in `clear_wr_session()`. Perhaps there should be a separate WebrecorderUtils class or something like that. Don't feel strongly about it either way :)

Reply:
We do actually have a number of link methods related to WR, for logic that is specific to a given link:
```
link.wr_collection_slug
link.wr_iframe_prefix
link.init_replay_for_user
link.upload_to_wr
```
To me these seem of a kind. I'm sure we haven't managed to be totally consistent, but generally we try to put as much logic as possible in the models, and only create util functions when the logic doesn't match nicely with any particular Perma model, like  `get_wr_session` and `clear_wr_session`. 

I do see what you are saying about the desirability of keeping the `request.session` manipulation all in one spot to assist in proper cleanup. IMO keeping link-related logic within the link model is weightier.... Hmmmm. Maybe just popping any key that is prefixed with `wr_` in `clear_wr_session` would be sufficiently elegant?